clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -236,12 +254,20 @@ public boolean rename(Path src, Path dst) throws IOException {
         // https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1505
         LakeFSFileStatus srcStatus;
         srcStatus = getFileStatus(src);
-        if (!srcStatus.isDirectory()) {
-            return renameFile(srcStatus, dst);
+        boolean result;
+        if (srcStatus.isDirectory()) {
+            result = renameDirectory(src, dst);
+        } else {
+            result = renameFile(srcStatus, dst);
+        }
+        if (!src.getParent().equals(dst.getParent())) {
+            deleteEmptyDirectoryMarkers(dst.getParent());
+            createDirectoryMarkerIfEmptyDirectory(src.getParent());
```

Comment:
Cool, thanks, I think I get it now.

Hopefully this behaviour is like what S3A does.  (It does not make sense if you mix regular S3 / lakefs file API operations with S3A operations, because then you can be left with a random mix of directory markers.  As long as this is what S3A does we are in good company and that's OK.)